go/ngql/query.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package ngql
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+
+	"github.com/attic-labs/noms/go/d"
+	"github.com/attic-labs/noms/go/types"
+	"github.com/graphql-go/graphql"
+)
+
+const (
+	atKey          = "at"
+	countKey       = "count"
+	keyKey         = "key"
+	rootKey        = "Root"
+	sizeKey        = "size"
+	targetHashKey  = "targetHash"
+	targetValueKey = "targetValue"
+	tmKey          = "tm"
+	valueKey       = "value"
+	valuesKey      = "values"
+	vrKey          = "vr"
+)
+
+func constructQueryType(rootValue types.Value, tm typeMap) *graphql.Object {
```

Comment:
go gotcha!

typeMap needs to be a  pointer or mutations will get lost when the backing store grows/is reallocated.